### PR TITLE
Update SQL browser documentation

### PR DESCRIPTION
**Pull request by zhmaeff**
Related to MC-1370

### CODE REVIEW

File: docs/modules/tools/pages/sql-browser.adoc [hunk 1]
```diff
@@ -10,50 +10,67 @@ If a map contains domain objects, the classes for those objects must be on the c
...
+A generated `CREATE MAPPING` statement is inserted into the editor and can be adjusted to your needs.
 
-* If the map does not have a mapping, an autogenerated `CREATE MAPPING` statement is inserted into the editor. This statement maps the data structure to a table that Hazelcast can access.
-* If the map already has a mapping, an autogenerated `SELECT` statement is inserted into the editor.
+Click **Confirm & Run**.
 
-To execute a query, click
-**Execute Query**.
+After the successful `CREATE MAPPING` SQL statement execution,
+you should see your mapping in the **Queryable objects** tree view:
```
Comment (promanenko) [inline docs/modules/tools/pages/sql-browser.adoc:33]: could you please add an information about columns and indexes? 
And update the screenshot to contain columns of multiple types and indexes too?

**Review by promanenko — APPROVED** [review]

File: docs/modules/tools/pages/sql-browser.adoc [hunk 2]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
+image:ROOT:SQLBrowserMenu.png[SQL Browser Toolbar Icon]
+
 Click **SQL Browser** in the toolbar.
 
 image:ROOT:SQLBrowserWindow.png[SQL Browser Window]
 
-To query an existing map in the cluster, select the map's name from the dropdown:
+To query an existing map in the cluster, you have to connect it by running `CREATE MAPPING` SQL statement.
+This statement maps the data structure to a table that can be queried with Hazelcast SQL.
+It can be generated by cluster or inserted manually into the SQL editor.
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:21]:
```suggestion
You can type this statement into the editor, or you can generate it automatically.
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 3]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
+
+Click **Connect Maps** button
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:23]:
```suggestion
To generate the `CREATE MAPPING` statement automatically, click the *Connect Maps* button
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 4]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
+and select the map's name from the dropdown to generate `CREATE MAPPING` statement based on the map content:
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:24]:
```suggestion
and select an existing map from the dropdown:
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 5]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
+
+image:ROOT:SQLBrowserConnectMaps.png[SQL Browser Connect Maps]
+
+A generated `CREATE MAPPING` statement is inserted into the editor and can be adjusted to your needs.
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:28]:
```suggestion
If you need to change the generated statement, you can use the editor.
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 6]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
+
+NOTE: In some cases `CREATE MAPPING` statement can not be generated. The Possible reasons are: the map is empty, the map is stored in `NATIVE` format without indexes.
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:30]:
```suggestion
NOTE: You cannot generate a `CREATE MAPPING` statement for empty maps or maps that store entries in the `NATIVE` format without indexes.
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 7]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
 
-* If the map does not have a mapping, an autogenerated `CREATE MAPPING` statement is inserted into the editor. This statement maps the data structure to a table that Hazelcast can access.
-* If the map already has a mapping, an autogenerated `SELECT` statement is inserted into the editor.
+Click **Confirm & Run**.
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:32]:
```suggestion
Click *Confirm & Run*.
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 8]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
 
-To execute a query, click
-**Execute Query**.
+After the successful `CREATE MAPPING` SQL statement execution,
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:34]:
```suggestion
After executing the `CREATE MAPPING` statement,
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 9]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
+you should see your mapping in the **Queryable objects** tree view:
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:35]:
```suggestion
you should see your mapping in the *Queryable objects* tree view:
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 10]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
+image:ROOT:SQLBrowserQueryableObjects.png[SQL Browser Queryable Objects]
 
-After a successful query execution, you will see the results under **Query Results** in two formats:
+Under the mapping name, you can find columns and indexes with their types.
 
-* As a plain list, where the `__key` and `this` columns represent the map entries' keys and values.
-* As JSON, where the same data is represented in JSON format. 
+== Executing Queries
+
+To execute a query, click **Execute Query**.
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:43]:
```suggestion
To execute a query, click *Execute Query*.
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 11]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
+
+After a successful query execution, you will see the results under **Query Results**
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:45]:
```suggestion
When the query is executed, the results are displayed under *Query Results*:
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 12]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
 
 .Example query results
 image:ROOT:SQLBrowserQueryResultTab.png[A list of query results]
 
-== Exporting Query Results as a CSV file
+There are two results representation modes:
+
+* **Show first 1000 records**
+
+This mode limits **Query Results** to the first 1000 records returned by the query.
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:54]:
```suggestion
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 13]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
+
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:55]:
```suggestion
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 14]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
+* **Stream last 1000 records**
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:56]:
```suggestion
* *Stream last 1000 records*: Display only the last 1000 records returned by the streaming query. The most recent record is always on the top of the list.
+
NOTE: This mode is only for displaying the results of streaming SQL queries.
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 15]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
 
-You can use **Export as CVS** to export the query result as a CSV file in
-a format https://tools.ietf.org/html/rfc4180[RFC 4180].
+NOTE: Should be used for displaying the Streaming SQL query results.
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:58]:
```suggestion
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 16]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
 
-Here is an example JSON view of the same query results:
+In this mode **Query Results** show only the last 1000 records returned by query,
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:60]:
```suggestion
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 17]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
+and the last record is always on the top of the list.
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:61]:
```suggestion
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 18]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
 
-image:ROOT:SQLBrowserJSONView.png[JSON query results]
+== Exporting Query Results
 
-To copy
-the result to the clipboard, use the **Copy to Clipboard** button in the top right corner.
+You can use **Export** to export the query result as a **CSV** file in
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:65]:
```suggestion
You can use **Export** to export the query result as a CSV file in
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 19]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
+a format https://tools.ietf.org/html/rfc4180[RFC 4180] or as a **JSON** file.
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:66]:
```suggestion
the https://tools.ietf.org/html/rfc4180[RFC 4180 format] or as a JSON file.
```

File: docs/modules/tools/pages/sql-browser.adoc [hunk 20]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
 
 == Viewing Executed SQL Queries
 
-In the **History** tab, you can see the history of query execution. The history holds the last
-one hundred executed queries. Each row contains **Execution Time**,
-**Execution Status** and **Query** values. You can re-execute any query from the
-history by pushing a button in the **Execute** column.
+In the **History** tab, you can see the history of query execution.
+The history holds the last one hundred executed queries.
+Each row contains **Execution Time**, **Execution Status** and **Query** values.
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:72]: Can we describe what these fields are? What are the possible executions statuses for example?

We may want to put this information in a list or a table.

File: docs/modules/tools/pages/sql-browser.adoc [hunk 21]
```diff
@@ -10,50 +10,71 @@ If a map contains domain objects, the classes for those objects must be on the c
...
+You can re-execute any query from the history by pushing a button in the **Execute** column.
 
 image:ROOT:SQLBrowserHistoryTab.png[A history of executed queries]
 
-You can also use **Clear Query Result** to remove the query results from both the list and JSON views.
+You can also use **Clear Query Result** to remove the query results.
```
Comment (JakeSCahill) [inline docs/modules/tools/pages/sql-browser.adoc:77]:
```suggestion
To remove the query results, click *Clear Query Result*.
```

Users skim for information, so we should lead with the information they are looking for.

**Review by JakeSCahill — CHANGES_REQUESTED** [review]
Minor suggestions and a request for more details.

File: docs/modules/tools/pages/sql-browser.adoc (hunk 20 above)
Comment (zhmaeff) [reply docs/modules/tools/pages/sql-browser.adoc:69]: I think the screenshot below shows all the possible values for statuses and other columns. WDYT?

**Review by JakeSCahill — APPROVED** [review]

**Pull request merged**